ydb/library/yql/sql/v1/node.h

```diff
@@ -1152,11 +1152,39 @@ namespace NSQLTranslationV1 {
         TNodePtr Compression;
     };
 
+    struct TVectorIndexSettings {
+        enum class EDistance {
+              Cosine        /* "cosine" */
+            , Manhattan     /* "manhattan" */
+            , Euclidean     /* "euclidean" */
+        };
+
+        enum class ESimilarity {
+              Cosine        /* "cosine" */
+            , InnerProduct  /* "inner_product" */
+        };
+
+        enum class EVectorType {
+              Float         /* "float" */
+            , Uint8         /* "uint8" */
+            , Int8          /* "int8" */
+            , Bit           /* "bit" */
+        };
+
+        using TMetric = std::variant<std::monostate, EDistance, ESimilarity>;
+        TMetric Metric;
+        std::optional<EVectorType> VectorType;
+        std::optional<ui32> VectorDimension;
+
+        bool Validate(TContext& ctx) const;
+    };
+
     struct TIndexDescription {
         enum class EType {
             GlobalSync,
             GlobalAsync,
             GlobalSyncUnique,
+            GlobalVectorKmeansTree
```

Comment:
```suggestion
            GlobalVectorKmeansTree,
```
We can keep old style of enum

Reply:
Done